src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Makes the System.Runtime.CompilerServices.NonNullTypesAttribute available in every compilation.
+    /// </summary>
+    internal class InjectedNonNullTypesAttributeSymbol : InjectedAttributeSymbol
+    {
+        private ImmutableArray<CSharpAttributeData> _lazyCustomAttributes;
+        private SymbolCompletionState _state;
+
+        private InjectedNonNullTypesAttributeSymbol(
+            AttributeDescription description,
+            NamespaceSymbol containingNamespace,
+            CSharpCompilation compilation,
+            Func<CSharpCompilation, NamedTypeSymbol, DiagnosticBag, ImmutableArray<MethodSymbol>> getConstructors,
+            DiagnosticBag diagnostics)
+            : base(description, containingNamespace, compilation, getConstructors, diagnostics)
+        {
+        }
+
+        public static InjectedNonNullTypesAttributeSymbol Create(NamespaceSymbol containingNamespace)
+        {
+            return new InjectedNonNullTypesAttributeSymbol(AttributeDescription.NonNullTypesAttribute, containingNamespace, containingNamespace.DeclaringCompilation, makeNonNullTypesAttributeConstructor, new DiagnosticBag());
+
+            ImmutableArray<MethodSymbol> makeNonNullTypesAttributeConstructor(CSharpCompilation compilation, NamedTypeSymbol containingType, DiagnosticBag diagnostics)
+            {
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                Binder.ReportUseSiteDiagnostics(boolType, diagnostics, Location.None);
+
+                var boolWithAnnotations = TypeSymbolWithAnnotations.Create(boolType);
+                // PROTOTYPE constructor should save the parameter into a field (for users of reflection)
```

Comment:
>// PROTOTYPE constructor should save the parameter into a field (for users of reflection) [](start = 16, length = 89)

Is that necessary?

Reply:
Confirmed with Jared (GetCustomAttributes reflection API).

---
In reply to: [214405480](https://github.com/dotnet/roslyn/pull/29180#discussion_r214405480) [](ancestors = 214405480)